src/main/java/org/folio/orders/utils/PoLineCommonUtil.java

```diff
@@ -229,4 +237,25 @@ public static JsonObject verifyProtectedFieldsChanged(List<String> protectedFiel
 
     return objectFromStorage;
   }
+
+  public static void verifyOngoingFieldsChanged(JsonObject compPOFromStorageJson, CompositePurchaseOrder compPO) {
+    JsonObject ongoingJson = compPOFromStorageJson.getJsonObject("ongoing");
+    if (Objects.nonNull(ongoingJson) && Objects.nonNull(compPO.getOngoing())) {
+      JsonPathParser oldObject = new JsonPathParser(ongoingJson);
+      JsonPathParser newObject = new JsonPathParser(JsonObject.mapFrom(compPO.getOngoing()));
+      if (Boolean.TRUE.equals(compPO.getOngoing().getIsSubscription())) {
+        checkFieldsChanged(oldObject, newObject, NOT_EDITABLE_ONGOING_SUBSCRIPTION_FIELDS, WRONG_ONGOING_SUBSCRIPTION_FIELDS_CHANGED);
+      } else {
+        checkFieldsChanged(oldObject, newObject, NOT_EDITABLE_ONGOING_NOT_SUBSCRIPTION_FIELDS, WRONG_ONGOING_NOT_SUBSCRIPTION_FIELDS_CHANGED);
+      }
+    }
+  }
+
+  private static void checkFieldsChanged(JsonPathParser oldObject, JsonPathParser newObject, List<String> fields, ErrorCodes error) {
+    for (String field: fields) {
+      if (ObjectUtils.notEqual(oldObject.getValueAt(field), newObject.getValueAt(field))) {
+        throw new HttpException(400, error);
```

Comment:
I think we should pass field name when throw exception

Reply:
![image](https://github.com/folio-org/mod-orders/assets/90304030/d6a1f001-d5c5-40dd-a5b5-1cc72da2fadf)
Fixed.

Reply:
Thank you